src/main/java/org/roaringbitmap/buffer/ImmutableRoaringBitmap.java

```diff
@@ -1026,6 +1019,53 @@ public boolean equals(Object o) {
     return false;
   }
 
+  /**
+   * Returns true if the other bitmap has no more than tolerance bits
+   * differing from this bitmap. The other may be transformed into a bitmap equal
+   * to this bitmap in no more than tolerance bit flips if this method returns true.
+   *
+   * @param other the bitmap to compare to
+   * @param tolerance the maximum number of bits that may differ
+   * @return true if the number of differing bits is smaller than tolerance
+   */
+  public boolean isHammingSimilar(ImmutableRoaringBitmap other, int tolerance) {
+    final int size1 = highLowContainer.size();
+    final int size2 = other.highLowContainer.size();
+    int pos1 = 0;
+    int pos2 = 0;
+    int budget = tolerance;
+    while(budget >= 0) {
+      short key1 = this.highLowContainer.getKeyAtIndex(pos1);
```

Comment:
Let's rollback to the first version. We can keep this as a future possible improvement.